src/main/kotlin/deadline/application/CreateDeadline.kt

```diff
@@ -0,0 +1,25 @@
+package deadline.application
+
+import deadline.domain.Deadline
+import deadline.domain.DeadlineRepository
+import deadline.domain.Time
+import note.domain.NoteIdentifier
+
+class AlreadyConfiguredDeadline : Throwable()
+
+class CreateDeadline(private val repository: DeadlineRepository) {
+
+    /***
+     * Creates a new deadline and saves it to the DeadlineRepository
+     * @throws AlreadyConfiguredDeadline if a new deadline already exists
+     * @return The deadline instance that was saved inside the repository
+     */
+    fun create(noteIdentifier: NoteIdentifier, time: Time): Deadline {
+        if (repository.get(noteIdentifier) != null) {
```

Comment:
It would look better like `repository.get(noteIdentifier)?.let { throw AlreadyConfiguredDeadline() }`

Reply:
Gotcha! That's true!
I thought it for a while and I think that it would look easier with:

```kotlin
    fun create(id: NoteIdentifier, time: Time): Deadline {
        val deadline = repository.get(id)?.let { throw AlreadyConfiguredDeadline() } ?: Deadline(id, time)
        repository.save(deadline)
        return deadline
    }
```

Reply:
That looks fine! ✅